packages/api-graphql/src/Providers/AWSWebSocketProvider/index.ts

```
@@ -263,22 +263,43 @@ export abstract class AWSWebSocketProvider {
 								'message',
 								publishListener,
 							);
-
+						cleanup();
 						resolve();
 					}
 
-					if (data.erroredEvents && data.erroredEvents.length > 0) {
-						// TODO: handle errors
+					if (data.errors && data.errors.length > 0) {
+						const errorTypes = data.errors.map((error: any) => error.errorType);
+						cleanup();
+						reject(new Error(`Publish errors: ${errorTypes.join(', ')}`));
```

Does this mean we'll close the WS connection if there's a publish error? If so, I don't think that's the behavior we want here. Publish errors would ideally be surfaced at the call site, similar to `events.connect`, e.g.
```js
try {
  await channel.publish({ some: 'data' })
} catch (err) {
  console.log('publish error', err)
}
```
This way, WS publish is self contained / decoupled from `channel.subscribe`.

---

We will not close the WS connection if there's a publish error. However, there was a bug in the implementation causing `uncaught errors`, it should surface the publish errors at the call site now.